### PR TITLE
EAS: Disable CTA if user is missing permissions

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR uses the same permissions necessary to access the integration enroll page to enable the External Audit Storage CTA button.

Previously, users could see the CTA, click the button and be redirected to a 404 page (since the web app wouldn't setup the react router for the enroll page without permissions).

Now, the button will be disabled, and on hover will display the message `Insuficient permissions. Reach out to your Teleport administrator to request External Audit Storage permissions.`.

Screenshot from 2023-12-06 11-08-53

### CODE REVIEW

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx [hunk 1]
```diff
@@ -100,9 +87,91 @@ export const ExternalAuditStorageCta = () => {
...
+            horizontal: 'right',
+          }}
+          transformOrigin={{
+            vertical: 'bottom',
+            horizontal: 'right',
+          }}
+        >
+          <StyledOnHover>
+            <Text fontSize="1">
+              Insuficient permissions. Reach out to your Teleport administrator
```
Comment (avatus) [inline web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx:134]:
```suggestion
              Insufficient permissions. Reach out to your Teleport administrator
```

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx [hunk 2]
```diff
@@ -100,9 +87,91 @@ export const ExternalAuditStorageCta = () => {
...
+            horizontal: 'right',
+          }}
+        >
+          <StyledOnHover>
+            <Text fontSize="1">
+              Insufficient permissions. Reach out to your Teleport administrator
+              to request External Audit Storage permissions.
+            </Text>
+          </StyledOnHover>
+        </Popover>
```
[the remark this answers is not in the document]
Comment (mcbattirola) [reply web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx:138]: That's exactly what I was looking for, thank you :+1: 

The component won't render anything with an empty tooltip, so I could remove the second `if`, just show the hover and disable the button based on user permissions.

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx [hunk 3]
```diff
@@ -100,9 +87,91 @@ export const ExternalAuditStorageCta = () => {
...
+        size="medium"
+        event={CtaEvent.CTA_EXTERNAL_AUDIT_STORAGE}
+        mr={5}
+      >
+        Contact Sales
+      </ButtonLockedFeature>
+    );
+  }
+
+  const [anchorEl, setAnchorEl] = useState();
```
[the remark this answers is not in the document]
Comment (mcbattirola) [reply web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx]: not necessary anymore with `shared/HoverTooltip`

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx [hunk 4]
```diff
@@ -100,6 +87,41 @@ export const ExternalAuditStorageCta = () => {
...
 };
 
+function CtaButton(props: { featureEnabled: boolean; userHasAccess: boolean }) {
+  if (!props.featureEnabled) {
+    return (
+      <ButtonLockedFeature
+        height="32px"
+        size="medium"
+        event={CtaEvent.CTA_EXTERNAL_AUDIT_STORAGE}
+        mr={5}
```
Comment (gzdunek) [inline web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx:97]: Here we have `mr=5` and in the other case `mr=2` (so they are inconsistent). Can we remove them and specify the spacing as `gap={2}` on the element from line 77?

**Review by gzdunek — APPROVED** [review]

**Pull request merged**